developer-documentation.md

```diff
@@ -79,3 +79,332 @@ address of any Google account that you own, such as an `@gmail.com` account.
+          '200':
+            description: An updated feature
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/Feature'
+
+  components:
+    schemas:
+      Feature:
+        description: A feature
+        type: object
+        properties:
+          id:
+            type: integer
+          name:
+            type: string
+          live:
+            type: boolean
+            description: Some optional field
+        required:
+          - id
+          - name
+  ```
+</details>
+
+### Step 4: Generate the Code
+
+Run: `npm run openapi`
+
+<details>
+  <summary>Example (click to expand)</summary>
+  
+  #### api.yaml
+  ```yaml
+  paths:
+    /features/{feature_id}:
+      post:
+        summary: Update a feature by ID.
+        description: >-
+          Update a feature with the given ID.
+          More details about this here.
+        operationId: updateFeatureById
+        parameters:
+          - name: feature_id
+            in: path
+            description: Feature ID
+            required: true
+            schema:
+              type: integer
+              format: int64
+        requestBody:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Feature'
+        responses:
+          '200':
+            description: An updated feature
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/Feature'
+      get:
+        summary: Get a feature by ID.
+        description: >-
+          Get a feature by ID. More details about this here.
+          Also, can do more comments
+        operationId: getFeatureById
+        parameters:
+          - name: feature_id
+            in: path
+            description: Feature ID
+            required: true
+            schema:
+              type: integer
+              format: int64
+        responses:
+          '200':
+            description: An updated feature
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/Feature'
+
+  components:
+    schemas:
+      Feature:
+        description: A feature
+        type: object
+        properties:
+          id:
+            type: integer
+          name:
+            type: string
+          live:
+            type: boolean
+            description: Some optional field
+        required:
+          - id
+          - name
+  ```
+</details>
+
+### Step 5: Incorporate Into Backend
+
+Currently, the repository is configured to use the generated Python data models for the backend. *Once all routes are generated by OpenAPI, it would be wise to revisit using the controllers as well*
+
+- Open `main.py`
+- Locate the `api_routes` variable.
+- Add a route.
+  - In this example, it would be `Route(f'{API_BASE}/features/<int:feature_id>', features_api.FeaturesAPI)`.
```

Comment:
So, how is this different than how we write our python code on the server side before introducing OpenAPI?

What are the generated model classes for?

Reply:
> So, how is this different than how we write our python code on the server side before introducing OpenAPI?
> 
> What are the generated model classes for?

Ah,  I think I see it come together in your PR [5/5].